sectionproperties/pre/sections.py

```diff
@@ -293,224 +622,757 @@ def plot_geometry(self, ax=None, pause=True, labels=False, perimeter=False):
+        on the line or by one point on the line and a vector. Either 'point_j' or 'vector'
+        must be given. If 'point_j' is given, 'vector' is ignored.
 
-        import sectionproperties.pre.sections as sections
+        Returns a tuple of two lists each containing new Geometry instances representing the 
+        "top" and "bottom" portions, respectively, of the bisected geometry.
 
-        points = [[0, 0], [100, 0], [75, 50], [25, 50], [15, 10], [85, 10], [70, 40], [30, 40]]
-        facets = [[0, 1], [1, 2], [2, 3], [3, 0], [4, 5], [5, 6], [6, 7], [7, 4]]
-        holes = [[50, 25]]
-        control_points = [[5, 5]]
-        perimeter = [0, 1, 2, 3]
+        If the line is a vertical line then the "right" and "left" portions, respectively, are 
+        returned.
 
-        geometry = sections.CustomSection(
-            points, facets, holes, control_points, perimeter=perimeter
-        )
-        mesh = geometry.create_mesh(mesh_sizes=[2.0])
+        :param point_i: A tuple of *(x, y)* coordinates to define a first point on the line
+        :type point_i: Tuple[float, float]
 
-    ..  figure:: ../images/sections/custom_geometry.png
-        :align: center
-        :scale: 75 %
+        :param point_j: Optional. A tuple of *(x, y)* coordinates to define a second point on the line
+        :type point_j: Tuple[float, float]
 
-        Custom section geometry.
+        :param vector: Optional. A tuple or numpy ndarray of *(x, y)* components to define the line direction.
+        :type vector: Union[Tuple[float, float], numpy.ndarray]
 
-    ..  figure:: ../images/sections/custom_mesh.png
-        :align: center
-        :scale: 75 %
+        :return: A tuple of lists containing Geometry objects that are bisected about the 
+            infinite line defined by the two given points. The first item in the tuple represents
+            the geometries on the "top" of the line (or to the "right" of the line, if vertical) and
+            the second item represents the geometries to the "bottom" of the line (or 
+            to the "left" of the line, if vertical).
 
-        Mesh generated from the above geometry.
+        :rtype: Tuple[List[Geometry], List[Geometry]]
+
+        The following example splits a 200PFC section about the y-axis::
+
+            import sectionproperties.pre.sections as sections
+            from shapely.geometry import LineString
+
+            geometry = sections.channel_section(d=200, b=75, t_f=12, t_w=6, r=12, n_r=8)
+            right_geom, left_geom = geometry.split_section((0, 0), (0, 1))
+        """
+        top_geoms_acc = []
+        bottom_geoms_acc = []
+        for geom in self.geoms:
+            top_geoms, bottom_geoms = geom.split_section(point_i, point_j, vector)
+            top_geoms_acc += top_geoms
+            bottom_geoms_acc += bottom_geoms
+        return (top_geoms_acc, bottom_geoms_acc)
+
+
+    def offset_section_perimeter(self, amount: float = 0, resolution: float = 12):
```

Comment:
@connorferster, should this also be renamed to `offset_perimeter` like in Geometry class to be consistent, as every other geometry manipulation is named consistently between the two geometry classes??

(and all example updated if changed?)

Reply:
Agreed! Updated and fixed.